ocean_lib/models/erc721_factory.py

```diff
@@ -192,3 +192,14 @@ def get_token_address(self, tx_id: str):
         )
 
         return registered_event[0].args.newTokenAddress
+
+    def get_data_token_address(self, tx_id: str):
```

Comment:
you can remove this from here, right? since in erc721_token you are only using the one with the same name, as defined in erc721_token

Reply:
Actually @calina-c , I implemented this and it's not working because the ERC721Factory catches the events, not the ERC721 token. If I want to retrieve the logs, the list will be empty when the ERC721 token calls `get_event_logs`. I think we should keep both in ERC721Factory since we want to retrieve the address by events.

Reply:
![Screenshot from 2022-01-12 16-10-30](https://user-images.githubusercontent.com/50501033/149242835-60f1445b-2df5-463c-94a8-4f337c497715.png)

Reply:
With factory parameter it worked
![Screenshot from 2022-01-12 16-17-01](https://user-images.githubusercontent.com/50501033/149243459-f856fecf-5162-46bd-9e36-7f229f03c334.png)

Reply:
I added a new commit, take a look. You don't need the events, you have the list of tokens pre and post-deployment.